server/migrations/migrations.js

```diff
@@ -75,37 +97,103 @@ function createHumanResourcesDataCore(knex) {
     });
 }
 
-function deleteHumanResourcesDataCore(knex) {
-    console.log("Delete " + tableName_human_resource__core_dataset);
-    return knex.schema.dropTable(tableName_human_resource__core_dataset)
+function deleteTable(knex, tableName) {
+    console.log("Delete " + tableName);
+    return knex.schema.dropTable(tableName)
 }
 
 
 function parseDataFromCsv(knex, fileLocation) {
     console.log("Parse data from " + fileLocation);
 
-    let papaparse = require('papaparse');
-
     let data = fs.readFileSync(fileLocation, "utf8");
-    let parseResult = papaparse.parse(data, {
-        header: true,
-        beforeFirstChunk: function (chunk) {
-            //delete the last line ",,,,,,,,"
-            chunk = chunk.replace(/\r?\n?[^\r\n]*$/, "").replace(/\r?\n?[^\r\n]*$/, "");
-            //tranform everything to small letters and add _ instead of spaces
-            let rows = chunk.split(/\r\n|\r|\n/);
-            let headings = rows[0].toLowerCase();
-            headings = headings.split(' ').join('_');
-            headings = headings.replace('/', '_');
-            rows[0] = headings;
-            return rows.join("\r\n");
-        }
-    });
+    let parseResult = papaparse.parse(data, parseConfig);
     console.log("Read data from file " + fileLocation);
     return parseResult;
 }
 
 function insertData(knex, tableName, data) {
-    console.log("Insert Data into " + tableName);
-    return knex(tableName).insert(data);
+    return knex(tableName)
+        .insert(data).catch(error => console.log(error));
 }
+
+async function download(url, knex) {
+    var streamSplitPart1;
+
+    await new Promise((resolve, reject) => {
+        var writeStream = new stream.Writable();
+        var headings;
+        var first = true;
+        writeStream._write = function (chunk, encoding, done) {
+
+            let stringChunk = chunk.toString();
+            let rows = stringChunk.split(/\r\n|\r|\n/);
+            if (streamSplitPart1) {
+                let streamSplitPart2 = rows[0];
+                let test = streamSplitPart1 + streamSplitPart2;
+                rows.splice(0, 1, test);
+                stringChunk = rows.join("\r\n");
+            }
+
+            if (!rows[rows.length - 1].endsWith("\r\n")) {
+                streamSplitPart1 = rows[rows.length - 1];
+                rows.splice(rows.length - 1, 1);
+                stringChunk = rows.join("\r\n");
+            } else {
+                streamSplitPart1 = null;
+            }
+
+            let streamParseConfig = {
+                header: true,
+                beforeFirstChunk: function (chunk) {
```

Comment:
why do we have different parse configs here?

Reply:
Because one is downloading and inserting the data from the download stream and the other one is inserting the data after the file is fully read into the system. When downloading the gzip we also need to care about "cutting" the stream into correct pieces for parsing